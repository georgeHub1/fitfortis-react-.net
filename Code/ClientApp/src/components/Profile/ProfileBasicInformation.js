import React, { Component } from 'react';
import { Form, Select, Row, Col } from 'antd';
import { injectIntl } from 'react-intl';
import PropTypes from 'prop-types';
import CustomCard from '../Common/CustomCard.js';
import CustomHeaderTooltip from '../Common/CustomHeaderTooltip';
import styles from './styles.module.less';
import * as R from 'ramda';
import moment from 'moment';
import { monthNames } from '../../constants/profile';
import analyticId from '../../utils/AnalyticsAndAbTests.tsx';

const { Option } = Select;

class ProfileBasicInformation extends Component {
  displayName = ProfileBasicInformation.name;
  static propTypes = {
    basicInformation: PropTypes.array,
    setTabValue: PropTypes.func
  };
  static defaultProps = {
    basicInformation: [],
    setTabValue: () => true
  };

  constructor (...args) {
    super(...args);

    this.state = {
      countDaysInMonth: this.daysInMonth(
        new Date().getMonth(),
        new Date().getFullYear()
      ),
      age: '',
      year: undefined,
      month: undefined,
      day: undefined,
      account: this.props.account,
      profileLifeStyle: this.props.profileLifeStyle,
      profileBasicInformation: this.props.profileBasicInformation,
      profileLifeStyleSelected: {},
      profileBasicInformationSelected: {}
    };
  }

  componentDidUpdate  ({profileBasicInformation, profileLifeStyle, account}) {
    const { profileBasicInformation: profileBasicInformationProps, profileLifeStyle: profileLifeStyleProps, account: accountProps } = this.props;

    if (!R.equals(account, accountProps)) {
      this.setState({
        account: {...accountProps}
      });
      this.calculateAge();
    }

    if (!R.equals(profileBasicInformation, profileBasicInformationProps)) {
      this.setState({
        profileBasicInformation: [...profileBasicInformationProps]
      }, () => {
        this.updateData();
      });
    }
    if (!R.equals(profileLifeStyle, profileLifeStyleProps)) {
      this.setState({
        profileLifeStyle: [...profileLifeStyleProps]
      }, () => {
        this.updateData();
      });
    }
  }

  componentDidMount () {
    this.updateData();
    analyticId.firebaseAnalyticsLog('Profile_BasicInformation');
  }

  updateData () {
    const { account } = this.props;
    const dateOfBirth = account !== undefined ? account.dateOfBirth : undefined;

    if (dateOfBirth !== undefined) {
      const date = new Date(dateOfBirth);

      this.setState({
        year: date.getUTCFullYear(),
        month: moment().locale('en').month(date.getUTCMonth()).format('MMMM'),
        day: date.getUTCDate()
      });
      this.getSectionSelected('prof');
    }
    this.getSectionSelected('profileBasicInformation');
    this.getSectionSelected('profileLifeStyle');
    this.calculateAge();
  }

  daysInMonth = (month, year) => {
    return new Date(year, month + 1, 0).getDate();
  };

  getYears = () => {
    const year = [];
    const minYear = 1940;
    const maxYear = new Date().getFullYear();

    for (let i = 0; i <= maxYear - minYear; i++) {
      year[i] = i + minYear;
    }
    return year.reverse();
  };

  handleSelectYearMonth = (value, key) => {
    const { account, setTabValue } = this.props;
    const { dateOfBirth } = account;
    const preMonth = dateOfBirth !== undefined ? new Date(dateOfBirth).getMonth() : undefined;
    const monthValue
      = key === 'month' ? moment().locale('en').month(value).format('MM') : preMonth;

    const preYear = dateOfBirth !== undefined ? new Date(dateOfBirth).getFullYear() : undefined;
    const yearValue
      = key === 'year' ? value : preYear;

    const noValueYear = yearValue ? yearValue : new Date().getFullYear();
    const noValueMonth = monthValue
      ? monthValue
      : monthNames[new Date().getMonth()];

    const index = monthNames.findIndex(i => i === noValueMonth);
    const countDaysInMonth = this.daysInMonth(index, noValueYear);


    const days = countDaysInMonth
      ? countDaysInMonth
      : this.daysInMonth(noValueMonth, noValueYear);

  this.setState({ [key]: value, countDaysInMonth: days });

  setTimeout(() => {
    const {year, month, day} = this.state;

    if (year !== undefined && month !== undefined && day !== undefined) {
      const newDateOfBirth = `${year}-${moment().locale('en').month(month).format('MM')}-${day}`;

      setTabValue({
        account: {
          ...account,
          dateOfBirth: newDateOfBirth

        }
      });
      }
    }, 500);
  };


  getSectionSelected = section => {
    let sectionSelected = {};
    const sectionInStateSelected = `${section}Selected`;
    const pattern = elm => elm.isSelected === true;
    const sectionInState = this.state[section] || [];

    sectionInState.forEach((item, index) => {
      const defaultItem = section === 'profileLifeStyle' ? [item.values[0]] : [{name:undefined}];
      const selectedElem = R.filter(pattern)(item.values).length !== 0
        ? R.filter(pattern)(item.values)
        : defaultItem;

        sectionSelected = R.assoc(item.key, selectedElem[0].id)(sectionSelected);
    });

    this.setState({
      [sectionInStateSelected]: {...sectionSelected}
    });
  }

  handleProfileSelect = (section, key, value) => {
    const { setTabValue } = this.props;
    const sectionInState = `${section}Selected`;
    const data = R.clone(this.state[section]);
    const newStatesectionSelected = R.assoc(key, value)(this.state[sectionInState]);
    const updateElem = elem => {
      return elem.id === value
        ?  R.assoc('isSelected', true)(elem)
        : R.assoc('isSelected', false)(elem);
    };
    const sectionChanges = data.map((item, index) => {
      return item.key === key
        ? {
          ...item,
          values: R.map(updateElem)(item.values)
        }
        : item;
    });

    this.setState({
      [sectionInState]: {...newStatesectionSelected},
      [section]: [...sectionChanges]
    });

     setTabValue({
      [section]: [...sectionChanges]
    });
  };


  calculateAge = () => {
    const {
      intl: { formatMessage },
      account
    } = this.props;

    const dateOfBirth = account !== undefined ? account.dateOfBirth : undefined;

    /* TODO: needs refactoring*/
    if (dateOfBirth !== undefined) {
      const DOB = new Date(dateOfBirth);
      const today = new Date();
      const age = today.getTime() - DOB.getTime();
      const elapsed = new Date(age);
      const year = elapsed.getYear() - 70;
      const month = elapsed.getMonth();

      let ageTotal = '';

      if (today <= DOB) {
        ageTotal = formatMessage({ id: 'ProfileBasicInformation.ageDateError' });
      } else {
        if (year > 0) {
          ageTotal += `${year} ${(year > 1) ? `${formatMessage({ id: 'ProfileBasicInformation.ageYears' })} ` : `${formatMessage({ id: 'ProfileBasicInformation.ageYear' })} ` }`;
        }
        if (month > 0) {
          ageTotal += `${month} ${(month > 1) ? `${formatMessage({ id: 'ProfileBasicInformation.ageMonths' })}` : `${formatMessage({ id: 'ProfileBasicInformation.ageMonth' })}` }`;
        }

        if (year === 0 && month === 0) {
          ageTotal += `${formatMessage({ id: 'ProfileBasicInformation.ageLessThanMonth' })}`;
        }
      }
      this.setState({
        age: ageTotal
      });
    }
  }


  render () {
    const {
      intl: { formatMessage }
    } = this.props;
    const { countDaysInMonth } = this.state;
    const daysInMonth = Array.from(
      { length: countDaysInMonth },
      (e, i) => i + 1
    );
    const years = this.getYears();
    const { profileLifeStyleSelected, profileBasicInformationSelected } =  this.state;
    const profileBasicInformation = this.state.profileBasicInformation || [];
    const profileLifeStyle = this.state.profileLifeStyle || [];
    const {day, month, year} = this.state;

    return (
      <Form
        id='profileBasicInformationForm'
        hideRequiredMark={true}
        onSubmit={this.handleSave}
        className="mainProfileBasicInformation"
      >
        <CustomCard>
          <div className="cardHeaderProfile">
            {formatMessage({
              id: 'ProfileBasicInformation.basicInfoLabel'
            })}
          </div>
          <Row gutter={32}>
              <Col xs={24} lg ={12}>
              <div className={styles.BirthDateWrapper}>
                <div className={styles.cardLabelProfile}>
                <Form.Item
                     label={formatMessage({
                      id: 'ProfileBasicInformation.birthDateLabel'
                    })}
                     className={styles.customLabel}
                  >
                </Form.Item>
                </div>
                <div className={styles.BirthDateSelects}>
                  <Row gutter={16}>
                  <Col xs={{ span: 10 }} sm={{ span: 12 }} className="monthInput">
                    <Form.Item >
                    <Select
                      getPopupContainer={trigger => trigger.parentNode}
                      className={styles.profileSelect}
                      placeholder={formatMessage({
                        id: 'ProfileBasicInformation.ageMonth'
                      })}
                      value={month}
                      onChange={value => {
                        this.handleSelectYearMonth(value, 'month');
                        setTimeout(() => {
                          this.calculateAge();
                        }, 500);
                      }}
                    >
                      {monthNames.map((month, key) => (
                        <Option key={month} value={month}>
                          {formatMessage({ id: `ProfileBasicInformation.${month}` })}
                        </Option>
                      ))}
                    </Select>
                  </Form.Item>
                  </Col>
                  <Col xs={{ span: 6 }} sm={{ span: 5 }} className="ageDayInput">
                  <Form.Item>
                    <Select
                      getPopupContainer={trigger => trigger.parentNode}
                      className={styles.profileSelect}
                      value={day}
                      placeholder={formatMessage({
                        id: 'ProfileBasicInformation.ageDay'
                      })}
                      onChange={value => {
                        this.handleSelectYearMonth(value, 'day');
                        setTimeout(() => {
                          this.calculateAge();
                        }, 500);
                      }}
                    >
                      {daysInMonth.map(day => (
                        <Option key={day} value={day < 10 ? `0${day}` : day }>
                          {day}
                        </Option>
                      ))}
                    </Select>
                  </Form.Item>
                  </Col>
                  <Col xs={{ span: 8 }} sm={{ span: 7 }} className="YearInput">
                  <Form.Item>
                    <Select
                      value={year}
                      getPopupContainer={trigger => trigger.parentNode}
                      className={styles.profileSelect}
                      placeholder={formatMessage({
                        id: 'ProfileBasicInformation.ageYear'
                      })}
                      onChange={value => {
                        this.handleSelectYearMonth(value, 'year');
                        setTimeout(() => {
                          this.calculateAge();
                        }, 500);
                      }}
                    >
                      {years.map(year => (
                        <Option key={year} value={year}>
                          {year}
                        </Option>
                      ))}
                    </Select>
                  </Form.Item>
                  </Col>
                  </Row>
                </div>
              </div>
            </Col>
            <Col xs={24} lg ={12} className={styles.ageLabel}>
              <div className={styles.cardLabelProfile}>
                <Form.Item
                     label= {formatMessage({
                      id: 'ProfileBasicInformation.ageLabel'
                    })}
                     className={styles.customLabel}
                  >
                </Form.Item>
              </div>
              <div className={styles.cardAge}>
                {this.state.age}
              </div>
            </Col>
          </Row>

          <Row gutter={32} className="BasicInfoSecRow">
          {
                profileBasicInformation.map((groupbasicInformation, index) => {
                  let placeholder = '';

                  switch (groupbasicInformation.name) {
                    case 'Sex at birth':
                      placeholder = formatMessage({
                        id: 'ProfileBasicInformation.sexAtBirthPlaceholder'
                      });
                    break;
                    case 'Gender identity':
                      placeholder = formatMessage({
                        id: 'ProfileBasicInformation.genderIdentityPlaceholder'
                      });
                    break;
                    case 'Ethnic origin':
                      placeholder = formatMessage({
                        id: 'ProfileBasicInformation.ethnicPlaceholder'
                      });
                    break;
                    default:
                     placeholder = '';
                    break;
                  }

                  return (
                    <Col sm = {24} md={12} key={index}>
                      <Form.Item
                        label={
                        <CustomHeaderTooltip
                          header={groupbasicInformation.name}
                          description={groupbasicInformation.description}
                          encyclopediaId={groupbasicInformation.encyclopediaId}
                        /> }
                      >
                      <Select
                        getPopupContainer={ trigger => trigger.parentNode }
                        className={ styles.profileSelect }
                        placeholder={placeholder}
                        style={{ width: '100%' }}
                        value={ profileBasicInformationSelected[groupbasicInformation.key] }
                        onChange={value => {
                          this.handleProfileSelect('profileBasicInformation', groupbasicInformation.key, value);
                        }}
                      >
                        {
                          groupbasicInformation.values.map((item, index) => {
                            return (
                              <Option value={ item.id} key={ index }>
                                { item.name }
                              </Option>
                            );
                          })
                        }
                      </Select>
                      </Form.Item>
                    </Col>
                  );
                })
              }

            </Row>
        </CustomCard>
        <CustomCard>
            <div className="cardHeaderProfile">
            {formatMessage({
                      id: 'ProfileBasicInformation.lifestyleLabel'
            })}
            </div>
            <div className={ styles.wrapLifestype}>
            <Row gutter={32}>
              {
                profileLifeStyle.map((groupLifeStyle, index) => {
                  return (
                    <Col sm = {24} md={12} key={index}>
                      <Form.Item
                        label={
                          <CustomHeaderTooltip
                            header={groupLifeStyle.name}
                            description={groupLifeStyle.description}
                            encyclopediaId={groupLifeStyle.encyclopediaId}
                          />
                        }
                      >
                      <Select
                        getPopupContainer={ trigger => trigger.parentNode }
                        className={ styles.profileSelect }
                        style={{ width: '100%' }}
                        value={ profileLifeStyleSelected[groupLifeStyle.key] }
                        onChange={value => {
                          this.handleProfileSelect('profileLifeStyle', groupLifeStyle.key, value);
                        }}
                      >
                        {
                          groupLifeStyle.values.map((item, index) => {
                            return (
                              <Option value={ item.id } key={ index }>
                                { item.name }
                              </Option>
                            );
                          })
                        }
                      </Select>
                      </Form.Item>
                    </Col>
                  );
                })
              }

            </Row>
            </div>
        </CustomCard>
      </Form>
    );
  }
}
const WrappedProfileBasicInformation = Form.create({
  name: 'profileBasicInformation'
})(ProfileBasicInformation);


export default injectIntl(WrappedProfileBasicInformation);
