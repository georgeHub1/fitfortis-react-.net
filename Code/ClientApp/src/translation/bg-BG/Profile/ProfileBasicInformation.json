{
  "basicInfoLabel": "Основни данни",
  "birthDateLabel": "Дата на раждане",

  "sexAtBirthPlaceholder" : "Избери пол",

  "genderIdentityPlaceholder": "Избери полова идентичност",

  "ethnicPlaceholder": "Избери етнически произход",

  "ageLabel": "Възраст",
  "ageDay": "Ден",
  "ageDays": "Дни",
  "ageMonth": "Месец",
  "ageMonths": "Месеца",
  "ageYear": "Година",
  "ageYears": "Години",
  "ageDateError": "Не можете да изберете дата в бъдещето",
  "ageLessThanMonth": "по-малка от 1 месец",

  "lifestyleLabel": "Начин на живот",

  "January": "Януари",
  "February": "Февруари",
  "March": "Март",
  "April": "Април",
  "May": "Май",
  "June": "Юни",
  "July": "Юли",
  "August": "Август",
  "September": "Септември",
  "October": "Октомври",
  "November": "Ноември",
  "December": "Декември"
}
