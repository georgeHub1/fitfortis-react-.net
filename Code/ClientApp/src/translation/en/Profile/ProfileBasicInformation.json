{
  "basicInfoLabel": "Basic Information",
  "birthDateLabel": "Date of birth",

  "sexAtBirthPlaceholder" : "Select sex at birth",

  "genderIdentityPlaceholder": "Select gender identity",

  "ethnicPlaceholder": "Select ethnic origin",

  "ageLabel": "Age",
  "ageDay": "Day",
  "ageDays": "Days",
  "ageMonth": "Month",
  "ageMonths": "Months",
  "ageYear": "Year",
  "ageYears": "Years",
  "ageDateError": "You cannot select future date",
  "ageLessThanMonth": "less than 1 month",

  "lifestyleLabel": "Lifestyle",
  
  "January": "January",
  "February": "February",
  "March": "March",
  "April": "April",
  "May": "May",
  "June": "June",
  "July": "July",
  "August": "August",
  "September": "September",
  "October": "October",
  "November": "November",
  "December": "December"
}
