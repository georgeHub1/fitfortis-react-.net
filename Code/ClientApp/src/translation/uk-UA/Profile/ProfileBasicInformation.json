{
  "basicInfoLabel": "Базова інформація",
  "birthDateLabel": "День народження",

  "sexAtBirthPlaceholder" : "Виберіть стать",

  "genderIdentityPlaceholder": "Виберіть гендерну ідентичність",

  "ethnicLabel": "Етнічне походження",

  "ageLabel": "Вік",
  "ageDay": "День",
  "ageDays": "Днів",
  "ageMonth": "Місяць",
  "ageMonths": "Місяці",
  "ageYear": "Рік",
  "ageYears": "Роки",
  "ageDateError": "Ви не можете вибрати майбутню дату",
  "ageLessThanMonth": "менше 1 місяця",

  "lifestyleLabel": "Спосіб життя",

  "January": "Січень",
  "February": "Лютий",
  "March": "Березень",
  "April": "Квітень",
  "May": "Може",
  "June": "Червень",
  "July": "Липень",
  "August": "Серпень",
  "September": "Вересень",
  "October": "Жовтень",
  "November": "Листопад",
  "December": "Грудень"
}
