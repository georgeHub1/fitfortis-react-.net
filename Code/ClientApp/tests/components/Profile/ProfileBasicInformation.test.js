import React from 'react';
import renderer from 'react-test-renderer';

import ProfileBasicInformation from '../../../src/components/Profile/ProfileBasicInformation.js';

const Container = ProfileBasicInformation;

describe('ProfileBasicInformation', () => {
  it('renders correctly', () => {
    const wrapper = renderer
      .create(<Container/>)
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
