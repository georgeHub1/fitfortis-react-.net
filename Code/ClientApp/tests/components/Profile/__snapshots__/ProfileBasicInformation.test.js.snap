// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileBasicInformation renders correctly 1`] = `
<form
  className="ant-form ant-form-horizontal ant-form-hide-required-mark mainProfileBasicInformation"
  id="profileBasicInformationForm"
  onSubmit={[Function]}
>
  <div
    className="ant-card undefined customCardContainer ant-card-bordered"
    style={
      Object {
        "background": "rgba(255, 255, 255, 0.8)",
      }
    }
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="cardHeaderProfile"
      />
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -16,
            "marginRight": -16,
          }
        }
      >
        <div
          className="ant-col ant-col-xs-24 ant-col-lg-12"
          style={
            Object {
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
        >
          <div>
            <div>
              <div
                className="ant-row ant-form-item"
                style={Object {}}
              >
                <div
                  className="ant-col ant-form-item-control-wrapper"
                  style={Object {}}
                >
                  <div
                    className="ant-form-item-control"
                  >
                    <span
                      className="ant-form-item-children"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                className="ant-row"
                style={
                  Object {
                    "marginLeft": -8,
                    "marginRight": -8,
                  }
                }
              >
                <div
                  className="ant-col monthInput ant-col-xs-10 ant-col-sm-12"
                  style={
                    Object {
                      "paddingLeft": 8,
                      "paddingRight": 8,
                    }
                  }
                >
                  <div
                    className="ant-row ant-form-item"
                    style={Object {}}
                  >
                    <div
                      className="ant-col ant-form-item-control-wrapper"
                      style={Object {}}
                    >
                      <div
                        className="ant-form-item-control"
                      >
                        <span
                          className="ant-form-item-children"
                        >
                          <div
                            className="ant-select ant-select-enabled"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseDown={[Function]}
                            onMouseOut={[Function]}
                            onMouseUp={[Function]}
                            onTouchStart={[Function]}
                          >
                            <div
                              aria-autocomplete="list"
                              aria-controls="test-uuid"
                              aria-expanded={false}
                              aria-haspopup="true"
                              className="ant-select-selection
            ant-select-selection--single"
                              onKeyDown={[Function]}
                              role="combobox"
                              tabIndex={0}
                            >
                              <div
                                className="ant-select-selection__rendered"
                              />
                              <span
                                className="ant-select-arrow"
                                onClick={[Function]}
                                style={
                                  Object {
                                    "WebkitUserSelect": "none",
                                    "userSelect": "none",
                                  }
                                }
                                unselectable="on"
                              >
                                <i
                                  aria-label="icon: down"
                                  className="anticon anticon-down ant-select-arrow-icon"
                                >
                                  <svg
                                    aria-hidden="true"
                                    className=""
                                    data-icon="down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                    />
                                  </svg>
                                </i>
                              </span>
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="ant-col ageDayInput ant-col-xs-6 ant-col-sm-5"
                  style={
                    Object {
                      "paddingLeft": 8,
                      "paddingRight": 8,
                    }
                  }
                >
                  <div
                    className="ant-row ant-form-item"
                    style={Object {}}
                  >
                    <div
                      className="ant-col ant-form-item-control-wrapper"
                      style={Object {}}
                    >
                      <div
                        className="ant-form-item-control"
                      >
                        <span
                          className="ant-form-item-children"
                        >
                          <div
                            className="ant-select ant-select-enabled"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseDown={[Function]}
                            onMouseOut={[Function]}
                            onMouseUp={[Function]}
                            onTouchStart={[Function]}
                          >
                            <div
                              aria-autocomplete="list"
                              aria-controls="test-uuid"
                              aria-expanded={false}
                              aria-haspopup="true"
                              className="ant-select-selection
            ant-select-selection--single"
                              onKeyDown={[Function]}
                              role="combobox"
                              tabIndex={0}
                            >
                              <div
                                className="ant-select-selection__rendered"
                              />
                              <span
                                className="ant-select-arrow"
                                onClick={[Function]}
                                style={
                                  Object {
                                    "WebkitUserSelect": "none",
                                    "userSelect": "none",
                                  }
                                }
                                unselectable="on"
                              >
                                <i
                                  aria-label="icon: down"
                                  className="anticon anticon-down ant-select-arrow-icon"
                                >
                                  <svg
                                    aria-hidden="true"
                                    className=""
                                    data-icon="down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                    />
                                  </svg>
                                </i>
                              </span>
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="ant-col YearInput ant-col-xs-8 ant-col-sm-7"
                  style={
                    Object {
                      "paddingLeft": 8,
                      "paddingRight": 8,
                    }
                  }
                >
                  <div
                    className="ant-row ant-form-item"
                    style={Object {}}
                  >
                    <div
                      className="ant-col ant-form-item-control-wrapper"
                      style={Object {}}
                    >
                      <div
                        className="ant-form-item-control"
                      >
                        <span
                          className="ant-form-item-children"
                        >
                          <div
                            className="ant-select ant-select-enabled"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseDown={[Function]}
                            onMouseOut={[Function]}
                            onMouseUp={[Function]}
                            onTouchStart={[Function]}
                          >
                            <div
                              aria-autocomplete="list"
                              aria-controls="test-uuid"
                              aria-expanded={false}
                              aria-haspopup="true"
                              className="ant-select-selection
            ant-select-selection--single"
                              onKeyDown={[Function]}
                              role="combobox"
                              tabIndex={0}
                            >
                              <div
                                className="ant-select-selection__rendered"
                              />
                              <span
                                className="ant-select-arrow"
                                onClick={[Function]}
                                style={
                                  Object {
                                    "WebkitUserSelect": "none",
                                    "userSelect": "none",
                                  }
                                }
                                unselectable="on"
                              >
                                <i
                                  aria-label="icon: down"
                                  className="anticon anticon-down ant-select-arrow-icon"
                                >
                                  <svg
                                    aria-hidden="true"
                                    className=""
                                    data-icon="down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                    />
                                  </svg>
                                </i>
                              </span>
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col ant-col-xs-24 ant-col-lg-12"
          style={
            Object {
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
        >
          <div>
            <div
              className="ant-row ant-form-item"
              style={Object {}}
            >
              <div
                className="ant-col ant-form-item-control-wrapper"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control"
                >
                  <span
                    className="ant-form-item-children"
                  />
                </div>
              </div>
            </div>
          </div>
          <div>
            
          </div>
        </div>
      </div>
      <div
        className="ant-row BasicInfoSecRow"
        style={
          Object {
            "marginLeft": -16,
            "marginRight": -16,
          }
        }
      />
    </div>
  </div>
  <div
    className="ant-card undefined customCardContainer ant-card-bordered"
    style={
      Object {
        "background": "rgba(255, 255, 255, 0.8)",
      }
    }
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="cardHeaderProfile"
      />
      <div>
        <div
          className="ant-row"
          style={
            Object {
              "marginLeft": -16,
              "marginRight": -16,
            }
          }
        />
      </div>
    </div>
  </div>
</form>
`;
